### PR TITLE
changed Infura tests to simulated backend

### PR DESCRIPTION
**Pull request by innainu**
- Changed from using Infura to simulated backend for retriever tests

### CODE REVIEW

**Comment by petersng** [comment]
One thing that can be updated is that `go-ethereum` updated their simulated backend to do what we did as a hack here with `contractutils/simulated.go`.   Can you try updating `go-ethereum` and updating to use their code so we don't have to use our hacky code?  This can be in another branch/PR.

**Comment by innainu** [comment]
yeah i know, i didn't wanna make that change when you weren't here bc not sure you wanted to do that

File: pkg/retriever/retriever_test.go [hunk 1]
```diff
@@ -31,96 +31,148 @@ func TestFilterersEventList(t *testing.T) {
...
+	}
+	if !sort.IntsAreSorted(blockNumbers) {
+		t.Error("Events are not sorted")
+	}
+
 }
 
-// Check last events. TODO: make better test here w simulated backend
-// XXX(PN): Infura has been having problems, so commenting this out so builds work.
-// func TestLastEvents(t *testing.T) {
```
Comment (petersng) [inline pkg/retriever/retriever_test.go]: Same question as above, is this test invalid now?

File: pkg/retriever/retriever_test.go [hunk 2]
```diff
@@ -31,96 +31,148 @@ func TestFilterersEventList(t *testing.T) {
 	}
 }
 
-// TestEventCollection tests that events are being collected,
-// TODO: make better test here w simulated backend
-// XXX(PN): Infura has been having problems, so commenting this out so builds work.
```
Comment (petersng) [inline pkg/retriever/retriever_test.go]: Is this test invalid now? This was commented out due to Infura issues, but probably can be re-enabled.

File: pkg/retriever/retriever_test.go (hunk 1 above)
Comment (innainu) [reply pkg/retriever/retriever_test.go]: oops, I changed the name. It should be TestLastEvents() sry about that

File: pkg/retriever/retriever_test.go (hunk 2 above)
Comment (innainu) [reply pkg/retriever/retriever_test.go]: I just changed the name of the test to "TestRetrieveMethod" since it's not a test related to the event collector

**Pull request merged**